CUSTOM.md

```diff
@@ -0,0 +1,218 @@
+    - InvaderDeckSetup(params)
+      - params: table - contains data about game setup
+        - level: number - level of current adversary (only exists for Adversary call)
+        - deck: table - table containing current setup for invader deck
+          - values of 1, 2, 3 indicate random card of that stage
+          - values of W, M, J, S indicate specific stage I cards
+          - values of We, Me, Je, Se, C indicate specific stage II cards
+          - values of MW, JW, SW, MJ, SM, JS indicate specific stage III cards
+          - values of <anything above>* will have UI treat card as one stage lower (i.e. Prussia 2)
+        - return table - updated table for invader deck setup
+  - mapSetup - allows you to modify the pieces on a board
+    - mapSetup: boolean - set to true if you have MapSetup function
+    - mapSetupComplete: boolean - set to true at end MapSetup function
+    - MapSetup(params)
+      - params: table - contains data about game setup
+        - level: number - level of current adversary (only exists for Adversary call)
+        - pieces: 2D table - contain keys for each land number, and each land table has values for piece name
+        - guid: string - guid of the board being setup (only exists for Adversary call)
+        - extra: boolean - whether the current board is the extra board from setup variant
+      - return table - updated table for pieces setup
+  - postSetup - called after all other setup steps have completed
+    - postSetup: boolean - set to true if you have PostSetup function
+    - postSetupComplete: boolean - set to true at end PostSetup function
+    - PostSetup(params)
+      - params: table - contains data about game setup (only exists for Adversary call)
+        - level: number - level of current adversary
+        - other: table - contains data about other adversary
+          - exist: boolean - whether there is a second adversary
+          - level: number - other adversary level
+- Adversary Specific:
+  - difficulty: table - contain keys 0->6 representing the difficulty modifier for each adversary level
+  - fearCards: 2D table - contains keys 0->6 representing the fear cards modifiers for each adversary level. Each adversary level should contain keys 1->3 representing the change of fear cards in each part of the deck
+  - reminderSetup
+    - reminderSetup: boolean - set to true if you have ReminderSetup function
+    - ReminderSetup(params)
+      - params: table - contains data about game setup
+        - level: number - level of current adversary
+      - return table - note the guids return must be in adversary bag
+        - ravage: string - guid of ravage reminder, use "135124" for generic one
+        - build: string - guid of build reminder, use "a178fa" for generic one
+        - explore: string - guid of explore reminder, use "a5b6b3" for generic one
+  - adversaryUI
+    - hasUI: boolean - set to true if you have AdversaryUI function
```

Comment:
Doesn't all adversaries need this due escalation?

Reply:
it's highly recommend for all adversaries to have UI hook, but it's not required since you can always just look at the actual card itself